### PR TITLE
Add Doppler

### LINKED ISSUE #84: Doppler
Steps to add doppler support

- [x] add rotating disk phantom datafile to zeahub
- [x] add Doppler func and ops
- [x] create a simple example notebook to visualize

@wesselvannierop you already had a snippet lying around for this right?

### PR DESCRIPTION
**Pull request by wesselvannierop**
Closes #84

- Implements `zea.doppler.color_doppler` function using the autocorrelation method originally written by @BeatriceFederici 
- Implements an example notebook for Doppler.

## Summary by CodeRabbit

- **New Features**
  - Color Doppler processing: generate 2D velocity maps from I/Q ultrasound data with optional spatial smoothing.
  - Notebook animation utility: export file-based animations of image sequences with physical axis scaling, adjustable framerate, colormap, size and DPI.

- **Documentation**
  - Doppler module added to public API autosummary.

- **Tests**
  - Added tensor operation tests for axis-wise functions and correlation.
  - Notebook-parameter consistency checks and extra test-run logging.

### CODE REVIEW

File: docs/source/notebooks/pipeline/doppler_example.ipynb [hunk 1]
```diff
@@ -0,0 +1,358 @@
...
+   "source": [
+    "# Doppler Ultrasound"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "8dc39619",
+   "metadata": {},
+   "source": [
+    "In this notebook, we demonstrate how to process and visualize Doppler ultrasound data using the Zea library. Doppler ultrasound is a non-invasive imaging technique that measures the frequency shift of ultrasound waves reflected from moving objects, such as blood flow in vessels."
```
Comment (tristan-deep) [inline docs/source/notebooks/pipeline/doppler_example.ipynb:16]:
```suggestion
    "In this notebook, we demonstrate how to process and visualize Doppler ultrasound data using the `zea` library. Doppler ultrasound is a non-invasive imaging technique that measures the frequency shift of ultrasound waves reflected from moving objects, such as blood flow in vessels."
```

Comment (tristan-deep) [inline docs/source/notebooks/pipeline/doppler_example.ipynb:16]: i would mention that this is color doppler as well. just to clarify the type of doppler ultrasound we are using.

File: zea/doppler.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
+"""Doppler functions for processing I/Q ultrasound data."""
+
+import keras
+import numpy as np
+from keras import ops
+
+if keras.backend.backend() == "jax":
```
Comment (tristan-deep) [inline zea/doppler.py:7]: this is not that multi-backend friendly. can we not use ops.map or do a stacking and reshaping to define apply along axis ourselves? (and move it to tensor_ops).

File: zea/doppler.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    import jax.numpy as jnp
+
+    apply_along_axis = jnp.apply_along_axis
+    correlate = ops.correlate
+else:
+    apply_along_axis = np.apply_along_axis
+    correlate = np.correlate
```
Comment (tristan-deep) [inline zea/doppler.py:14]: why does ops.correlate need to be here?

File: zea/doppler.py [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+def iq2doppler(
+    data,
+    center_frequency,
+    pulse_repetition_frequency,
+    sound_speed,
+    hamming_size=None,
+    lag=1,
+):
+    """Compute Doppler from packet of I/Q Data.
```
Comment (tristan-deep) [inline zea/doppler.py:25]:
```suggestion
    """Compute Color Doppler from packet of I/Q Data.
```

File: zea/doppler.py [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+    Args:
+        data (ndarray): I/Q complex data of shape (grid_size_z, grid_size_x, n_frames).
```
Comment (tristan-deep) [inline zea/doppler.py:28]: would it be better to have the [standard zea data dim order](https://zea.readthedocs.io/en/latest/data-acquisition.html) here with n_frames in the front? That would lead to less confusion i feel when people are using this function alongside a pipeline. In your example I also see you do a moveaxis in the end.

**Review by tristan-deep — CHANGES_REQUESTED** [review]

File: zea/doppler.py (hunk 2 above)
Comment (wesselvannierop) [reply zea/doppler.py:7]: Yep, it is a shame keras has not implemented it :(
